src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -160,20 +165,23 @@ class WidgetQueries extends React.Component<Props, State> {
           return {
             ...prevState,
             results,
+            error: false,
+            errorMessage: undefined,
             loading: completed === promises.length ? false : true,
           };
         });
-      } catch (e) {
-        this.setState({error: true});
+      } catch (err) {
+        const errorMessage = err?.responseJSON?.detail || t('An unknown error occurred.');
+        this.setState({error: true, errorMessage});
```

Comment:
Do we need both the boolean and the message? Couldn't use the presence of the message to imply that there is an error?

Reply:
I removed `error` in https://github.com/getsentry/sentry/pull/22949/commits/822fb65d34d6e03381e0a79e4549812d3bd6483b